.github/workflows/cmake.yml

```
@@ -0,0 +1,90 @@
+
+env:
+    CMAKE_UNITY_BUILD: OFF
+
+permissions:
+    contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+    build-cmake:
+        runs-on: ubuntu-20.04
+        env:
+            CMAKE_OPTIONS: -DPython_LOOKUP_VERSION=3.8 -DUSE_CCACHE=ON
+            cache-name: cmake-ubuntu-20.04
+        steps:
+            - name: Checkout GRASS
+              uses: actions/checkout@v4.1.1
+            - name: Setup cache
+              uses: actions/cache@v4.0.2
+              id: cache
+              with:
+                  path: ${{ github.workspace }}/.ccache
+                  key: "${{ env.cache-name }}-${{ runner.os }}-${{ github.base_ref }}\
+                        ${{ github.ref_name }}-${{ github.run_id }}"
+                  restore-keys: |
+                    ${{ env.cache-name }}-${{ runner.os }}-${{ github.base_ref }}
+                    ${{ env.cache-name }}-${{ runner.os }}
+            - name: Install CMake 3.16.0
+              run: |
+                cd ${GITHUB_WORKSPACE}
+                wget https://github.com/Kitware/CMake/releases/download/v3.16.0/cmake-3.16.0-Linux-x86_64.tar.gz
+                tar xzf cmake-3.16.0-Linux-x86_64.tar.gz
+                echo "CMAKE_DIR=$GITHUB_WORKSPACE/cmake-3.16.0-Linux-x86_64/bin" >> $GITHUB_ENV
+                echo "$GITHUB_WORKSPACE/cmake-3.16.0-Linux-x86_64/bin" >> $GITHUB_PATH
+            - run: |
+                cmake --version
+            - name: Install dependencies
+              run: |
+                sudo add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+                sudo apt-get update
+                sudo apt-get install -y -q build-essential \
+                 flex make bison gcc libgcc1 g++ ccache \
+                 python3 python3-dev \
+                 python3-opengl python3-wxgtk4.0 \
+                 python3-dateutil libgsl-dev python3-numpy \
+                 wx3.0-headers wx-common libwxgtk3.0-gtk3-dev \
+                 libwxbase3.0-dev libbz2-dev zlib1g-dev gettext \
+                 libtiff5-dev libpnglite-dev libcairo2 libcairo2-dev \
+                 sqlite3 libsqlite3-dev \
+                 libpq-dev libreadline6-dev libfreetype6-dev \
+                 libfftw3-3 libfftw3-dev \
+                 libboost-thread-dev libboost-program-options-dev libpdal-dev\
+                 subversion libzstd-dev checkinstall \
+                 libglu1-mesa-dev libxmu-dev \
+                 ghostscript wget -y make ccache g++
+            - run: python -m pip install ninja
+            - name: Configure ccache
+              run: |
+                  echo CCACHE_BASEDIR=$PWD >> ${GITHUB_ENV}
+                  echo CCACHE_DIR=$PWD/.ccache >> ${GITHUB_ENV}
+                  echo CCACHE_MAXSIZE=250M >> ${GITHUB_ENV}
+                  ccache -z
+              working-directory: ${{ github.workspace }}
+            - name: Configure
+              run: |
+                cmake ${CMAKE_OPTIONS} -S $GITHUB_WORKSPACE -B $GITHUB_WORKSPACE/build
```

I made a pr of this branch to my fork, then made changes of the merged branch as a new PR, to have CI available.

I debugged a bit, and had to specify the generator since it didn't choose it, it still chose make, even with the latest cmake 3.29.2.

So, 
```suggestion
                cmake ${CMAKE_OPTIONS} -S $GITHUB_WORKSPACE -B $GITHUB_WORKSPACE/build -G Ninja
```